signature-specification.md

```diff
@@ -0,0 +1,218 @@
+1. Descriptor MUST contain `mediaType`, `digest`, `size` fields.
+2. Descriptor MAY contain `annotations`. The `annotations` are being used to store signed attributes.
+3. Descriptor MAY contain `artifactType` field for artifact manifests, or the `config.mediaType` for `oci.image` based manifests.
+
+Examples:
+```json
+{
+   "mediaType": "application/vnd.oci.image.manifest.v1+json",
+   "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+   "size": 16724
+}
+```
+
+```json
+{
+   "mediaType": "sbom/example",
+   "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+   "size": 32654
+}
+```
+
+### Signed Attributes
+Notary v2 requires the signature envelope to support the following signed attributes:
+* **Creation time**: The time at which the signature was generated. Its value should be numeric representing the number of seconds(not milliseconds) since Epoch.
+* **Expiration time**: The time after which signature shouldn't be considered valid. Its value should be numeric representing the number of seconds since Epoch. This is an OPTIONAL attribute.
+
+### Unsigned Attributes
+* **Certificates**: The ordered collection of X.509 certificates with a signing certificate as the first entry.
+* **Timestamp**:  The time stamp token generated for a given signature. Only [RFC3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2) compliant TimeStampToken are supported. This is an OPTIONAL attribute.
+
+### Supported Signature Envelopes
+#### JWS JSON Serialization
+In JWS JSON Serialization([RFC7515](https://datatracker.ietf.org/doc/html/rfc7515)), data is stored as either claims or headers(protected and unprotected). Notary v2 uses JWS JSON Serialization for the signature envelope with some additional constraints on the structure of claims and headers.
+
+Unless explicitly specified as OPTIONAL, all fields are required. Also, there shouldn’t be any additional fields other than ones specified in JWSPayload, ProtectedHeader, and UnprotectedHeader.
+
+**JWSPayload a.k.a. Claims**:
+Notary v2 is using one private claim(notary) and two public claims(iat and exp). An example of the claim is described below
+```json
+{
+   "notary": {  
+       "subject": {
+           "mediaType": "application/vnd.oci.image.manifest.v1+json",
+           "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+           "size": 16724,
+           "annotations": {
+              "key1": "value1",
+              "key2": "value2",
+               ...
+           }
+       }
+   },
+   "iat": 1234567891000,
+   "exp": 1234567891011
+}
+```
+The payload contains the subject manifest and other attributes that have to be integrity protected.
+
+* **`notary`**(*string-string map*): This is a REQUIRED top-level node and a private claim encapsulating the notary v2 data.
+* **`subject`**(*descriptor*): A REQUIRED manifest that needs to be integrity protected.
```

Comment:
Do you need to document this again, it was already covered in Payload section, with artifactType etc. being optional.